src/adapter/src/coord/sequencer/cluster.rs

```diff
@@ -450,9 +457,32 @@ impl Coordinator {
             },
         };
 
-        let id = self.catalog_mut().allocate_user_replica_id().await?;
+        let cluster = self.catalog().get_cluster(cluster_id);
+
+        // Only internal users can add replicas to managed clusters
+        if cluster.is_managed() && !session.user().is_internal() {
+            // TODO: PlanError isn't great to use here.
```

Comment:
Can this go in plan_create_cluster_replica?

Reply:
Unfortunately not because there we don't have access to `session.user()` as far as I can tell.